Optimizer/FcTypeChecker.hs

```diff
@@ -219,19 +212,336 @@ ensureIdenticalTypes types = unless (go types) $ throwError "Type mismatch in ca
     go []       = True
     go (ty:tys) = all (eqFcTypes ty) tys
 
+ensureMatchingArgTypes :: [FcType] -> [FcType] -> FcM ()
```

Comment:
Add a comment. It's not clear what this function does.
What are `rand_tys` and `arg_tys`?